src/memray/reporters/tui.py

```diff
@@ -1,83 +1,163 @@
+import contextlib
+import dataclasses
 import os
+import pathlib
+import sys
+import threading
 from collections import defaultdict
 from collections import deque
-from copy import deepcopy
 from dataclasses import dataclass
 from datetime import datetime
+from functools import total_ordering
 from math import ceil
+from typing import Any
 from typing import DefaultDict
 from typing import Deque
 from typing import Dict
 from typing import Iterable
 from typing import List
 from typing import Optional
-from typing import Sequence
 from typing import Set
 from typing import Tuple
+from typing import cast
 
-from rich.layout import Layout
 from rich.markup import escape
-from rich.panel import Panel
-from rich.progress_bar import ProgressBar
-from rich.table import Column
-from rich.table import Table
+from rich.segment import Segment
+from rich.style import Style
+from rich.text import Text
+from textual import events
+from textual import log
+from textual.app import App
+from textual.app import ComposeResult
+from textual.binding import Binding
+from textual.color import Color
+from textual.color import Gradient
+from textual.containers import Container
+from textual.containers import HorizontalScroll
+from textual.dom import DOMNode
+from textual.message import Message
+from textual.reactive import reactive
+from textual.screen import Screen
+from textual.strip import Strip
+from textual.widget import Widget
+from textual.widgets import DataTable
+from textual.widgets import Footer
+from textual.widgets import Label
+from textual.widgets import Static
+from textual.widgets.data_table import RowKey
 
 from memray import AllocationRecord
+from memray import SocketReader
 from memray._memray import size_fmt
 
 MAX_MEMORY_RATIO = 0.95
 
 DEFAULT_TERMINAL_LINES = 24
```

Comment:
Is this used? Seems that we copied it to summary.py

Reply:
Used only for type annotations.

Reply:
oh, wait, I misread - I thought you were asking about the highlighted line in the diff, not the last line in it. Yep, that's unused now.